File: maratona/L.py
def fatorial(n):
    f = 1
    while (n > 0):
        f *= n
        n -= 1
    print(f)
    return f

zeros = 0
n = int(input())
fat = fatorial(n)

temp = str(fat)

i = len(temp)-1

while (i >= 0):
    print(temp[i])
    if (temp[i] == '0'):
        zeros += 1
    else:
        break

    i -= 1

print(zeros)

File: maratona/L.cpp
#include <bits/stdc++.h>
using namespace std;

long long fatorial(long long n){
    long long f = 1;
    while (n > 0){
        f *= n;
        n--;
    }
    return f;
}

int main(){
    long long n, zeros=0;
    long long fat;
    cin >> n;
    fat = fatorial(n);
    string temp = to_string(fat);

    for (long long i = temp.length()-1; i >= 0; i--){
        if (temp[i] == '0'){
            zeros++;
        } else {
            break;
        }
    }

    cout << zeros << endl;

    return 0;
}